packages/slate-editor/src/lib/hooks/useImage.ts

```diff
@@ -9,13 +10,21 @@ type OriginalURIOrBlobDataURI = string;
 interface State {
     error?: Error;
     loading: boolean;
-    progress: number;
+    progress: number | undefined;
     loaded: boolean;
     url: OriginalURIOrBlobDataURI | undefined;
 }
 
 export function useImage(src: string): State {
     const fetchImage = useCallback(() => fetchImageWithProgress(src), [src]);
     const { error, loading, progress, value } = useAsyncProgress(fetchImage);
-    return { error, loading, progress, loaded: Boolean(value), url: value };
+    return {
+        error,
+        loading,
+        // We can't track progress for unsupported origins and this way we can rely
+        // on the <ImageWithLoadingPlaceholder> component to estimate the progress.
+        progress: isSupportedImageOrigin(src) ? progress : undefined,
```

Comment:
It's not possible to make the progress reporting work due to CORS so I am returning `undefined` here for unsupported origins so we get at least an approximation in the image component.